subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/groovy/build.gradle

```diff
@@ -20,16 +32,16 @@ abstract class IvyVariantDerivationRule implements ComponentMetadataRule {
 
         context.details.addVariant("runtimeElements", "default") {
             attributes {
-                attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, getObjects().named(LibraryElements, LibraryElements.JAR))
-                attribute(Category.CATEGORY_ATTRIBUTE, getObjects().named(Category, Category.LIBRARY))
-                attribute(Usage.USAGE_ATTRIBUTE, getObjects().named(Usage, Usage.JAVA_RUNTIME))
+                attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, jarLibraryElements)
```

Comment:
Unfortunately this fails when running the snippet tests. The code attempts to find `jarLibraryElements` in the wrong scope.

Reply:
I'll try and reproduce this from Roberto's fork.

Reply:
@rpalcolea you can reproduce locally with this invocation:
`$ ./gradlew :docs:docsTest --tests "*snippet-dependency-management-customizing-resolution-ivy-metadata-rule*" --no-configuration-cache`

Reply:
@rpalcolea please apply this patch and the sample will work:

```
diff --git a/subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/groovy/build.gradle b/subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/groovy/build.gradle
index 4a552b9520e..90f86e8d2bd 100644
--- a/subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/groovy/build.gradle
+++ b/subprojects/docs/src/snippets/dependencyManagement/customizingResolution-ivyMetadataRule/groovy/build.gradle
@@ -11,10 +11,10 @@ repositories {
 
 // tag::ivy-component-metadata-rule[]
 abstract class IvyVariantDerivationRule implements ComponentMetadataRule {
-    private LibraryElements jarLibraryElements
-    private Category libraryCategory
-    private Usage javaRuntimeUsage
-    private Usage javaApiUsage
+    final LibraryElements jarLibraryElements
+    final Category libraryCategory
+    final Usage javaRuntimeUsage
+    final Usage javaApiUsage
 
     @Inject
     IvyVariantDerivationRule(ObjectFactory objectFactory) {
```

Reply:
Alright, thanks for sharing this. PR is updated :) @DPUkyle